src/common/disposable.ts

```diff
@@ -2,14 +2,17 @@
 // Licensed under the MIT license.
 
 export class Disposable {
-    private disposed = false;
-    constructor(private callOnDispose: () => any) { }
+    #disposed = false;
+    #callOnDispose: () => any;
+
+    constructor(callOnDispose: () => any) {

```

Comment:
Can client pass in null or undefined to the constructor? it's all good if no.

Reply:
Technically, it is possible. Using "any" or optional type can pass the static type check. It would be better to validate `callOnDispose`'s type here.

Reply:
The answer is no. @juniwang 
()=>any is a function whose return type is any. Undefined or null cannot be converted to a function type. @zhiyuanliang-ms